vaxrank/report.py

```diff
@@ -76,37 +86,101 @@ def ascii_report_from_ranked_vaccine_peptides(
+                'aa_mutant': aa_mutant,
+                'aa_after_mutation': aa_after_mutation,
+                'score': vaccine_peptide.combined_score,
+                'length': len(amino_acids),
+                'expression_score': vaccine_peptide.expression_score,
+                'mutant_epitope_score': vaccine_peptide.mutant_epitope_score,
+                'wildtype_epitope_score': vaccine_peptide.wildtype_epitope_score,
+                'num_alt_reads_supporting_protein_sequence':
+                    mutant_protein_fragment.n_alt_reads_supporting_protein_sequence,
+                'num_mutant_amino_acids':
+                    mutant_protein_fragment.n_mutant_amino_acids,
+                'mutation_distance_from_edge':
+                    mutant_protein_fragment.mutation_distance_from_edge,
+                'epitopes': epitopes,
+            }
+            
+            # compile epitope info
             for epitope_prediction in vaccine_peptide.epitope_predictions:
                 if epitope_prediction.overlaps_mutation and epitope_prediction.ic50 <= 2000:
-                    lines.append("\t\t\t * %s %0.2f (%s)" % (
-                        epitope_prediction.peptide_sequence,
-                        epitope_prediction.ic50,
-                        epitope_prediction.allele))
-    return "\n".join(lines)
+                    epitope_dict = {
+                        'sequence': epitope_prediction.peptide_sequence,
+                        'ic50': epitope_prediction.ic50,
+                        'allele': epitope_prediction.allele,
+                    }
+                    epitopes.append(epitope_dict)
+
+            peptides.append(peptide_dict)
+        variants.append(variant_dict)
+
+    template_data['variants'] = variants
+    return template_data
+
+
+def make_ascii_report_from_ranked_vaccine_peptides(
+        ranked_variants_with_vaccine_peptides,
+        mhc_alleles,
+        variants,
+        bam_path,
+        ascii_report_path):
+    template = JINJA_ENVIRONMENT.get_template('templates/template.txt')
+    template_data = compute_template_data(
+        ranked_variants_with_vaccine_peptides,
+        mhc_alleles,
+        variants,
+        bam_path)
+    report = template.render(template_data)
+    print(report)
+    with open(ascii_report_path, "w") as f:
+        f.write(report)
+
+
+# this is the hackiest thing in all the hacks. once we find out what the useful
+# vaxrank output formats are (will PDF be used? viewed? printed?, we can
+# fix this if necessary
+def compute_pdf_length_from_template_data(template_data):
+    length = 8  # 8" baseline from patient info header
+    variants = template_data['variants']
+    for variant in variants:
+        length += 6  # 6" space for variant description
+        for peptide in variant['peptides']:
+            length += 4  # 4" space for peptide description
+    return length
+
+
+def make_html_report_from_ranked_vaccine_peptides(
```

Comment:
Could this be split into make_pdf_report… (which I guess would just call make_html_… making a tmp html report, and then convert it to PDF, then remove the html file); would be nice to have them all split up.